h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -292,12 +292,15 @@ to a map (possibly the metadata map) -
         Object o = config.get("compress");
         this.compressionLevel = o == null ? 0 : (Integer) o;
         String fileName = (String) config.get("fileName");
-        o = config.get("pageSplitSize");
-        if (o == null) {
-            pageSplitSize = fileName == null ? 4 * 1024 : 16 * 1024;
-        } else {
-            pageSplitSize = (Integer) o;
+        fileStore = (FileStore) config.get("fileStore");
+        fileStoreIsProvided = fileStore != null;
+        if(fileStore == null && fileName != null) {
+            fileStore = new FileStore();
         }
+        o = config.get("pageSplitSize");
+        pageSplitSize = o != null         ? (Integer) o :
+                        fileStore == null ? 48 :
+                                            16 * 1024;
```

Comment:
I'm not a huge fan of the ternary operator, IMO it makes code harder to read, but one of them at a time is OK.